pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -123,6 +123,13 @@ public class PersistentDispatcherMultipleConsumers extends AbstractDispatcherMul
     protected final ExecutorService dispatchMessagesThread;
     private final SharedConsumerAssignor assignor;
 
+    /**
+     * Delivery was paused at least once in the earlier time, due to the cursor data can not fully persist.
+     * Note: do not use this field to confirm whether the delivery should be paused,
+     *       please call {@link #shouldPauseOnAckStatePersist}.
+     */
+    private volatile boolean pausedDueToCursorDataCanNotFullyPersist = false;
```

Comment:
We have two parts will update this field

1. readMoreEntries()
2. afterAckMessages()

But they are different threads? even if we have `synchronized` for the `readMoreEntries()` method. But the `afterAckMessages()` method will not acquire the same lock?

Reply:
And also I think the `pausedDueToCursorDataCanNotFullyPersist`  should be named something like `BLOCKED_DISPATCHER_ON_CURSOR_DATA_NOT_FULLY_PERSIST`
because we have defined `BLOCKED_DISPATCHER_ON_UNACKMSG`

Reply:
@Technoboy- 

> And also I think the pausedDueToCursorDataCanNotFullyPersist should be named something like BLOCKED_DISPATCHER_ON_CURSOR_DATA_NOT_FULLY_PERSIST
because we have defined BLOCKED_DISPATCHER_ON_UNACKMSG

Sure, renamed `pausedDueToCursorDataCanNotFullyPersist` to `blockedDispatcherOnCursorDataCanNotFullyPersist`